### PR TITLE
Add state-store support

### PR DESCRIPTION
**Pull request by nivemaham**
add state-store check to validate against spoofing

### CODE REVIEW

File: authorizer-app-backend/src/main/java/org/radarbase/authorizer/util/StateStore.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    data class State(val uuid: String, val sourceType: String) {
+        override fun toString() = "state=$uuid&sourceType=$sourceType"
+        companion object {
+            fun toState(string: String) : State {
+                val map = string.split("&")
+                return State(map[0].split("=")[1], map[1].split("=")[1])
+            }
+        }
```
Comment (blootsvoets) [inline authorizer-app-backend/src/main/java/org/radarbase/authorizer/util/StateStore.kt:54]: Does this work, with additional parameters inside a single parameter?

**Review by blootsvoets — APPROVED** [review]
LGTM. Not sure whether joining two parameters to a state works, but if it's tested I'm all for it.

File: authorizer-app-backend/src/main/java/org/radarbase/authorizer/util/StateStore.kt (hunk 1 above)
Comment (nivemaham) [reply authorizer-app-backend/src/main/java/org/radarbase/authorizer/util/StateStore.kt:54]: I have tested the above implementation. e.g. For the above `toString()`, the `toState()` works fine. If we need to pass additional parameters in the future, maybe we can opt for more complicated solution.

**Comment by blootsvoets** [comment]
@nivemaham additional consideration: why not encode the source type in the path As a path segment, and in the state only pass a UUID, which can also be the ID referencing more parts of the state?

**Pull request merged**